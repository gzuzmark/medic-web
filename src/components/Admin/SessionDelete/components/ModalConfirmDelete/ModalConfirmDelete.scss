.ModalConfirmDelete {
  display: flex;
  flex-direction: column;
  flex-grow: 1;
  &_texts {
    margin: auto 0;
    text-align: center;
  }
  &_error {
    text-transform: uppercase;
    svg {
      fill: rgb(255, 131, 137);
      width: 22px;
      height: 22px;
      display: inline-block;
      margin-bottom: -7px;
    }
  }
  &_buttons {
    display: flex;
    justify-content: space-around;
    margin: 0 auto;
    width: 60%;
  }
}
