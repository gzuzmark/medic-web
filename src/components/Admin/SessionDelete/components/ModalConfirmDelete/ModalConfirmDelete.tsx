import * as React from 'react';
import ConfirmButtons from "../../../../../common/ConfirmButtons/ConfirmButtons";
import ConsoleModalConfirm from "../../../../../common/ConsoleModal/ConsoleModalConfirm";
import {Title2} from "../../../../../common/ConsoleText";
import Icon from "../../../../../common/Icon/Icon";
import './ModalConfirmDelete.scss';


interface IPropsModalConfirmDelete {
    show: boolean;
    styles?: React.CSSProperties;
    disabled: boolean;
    loading: boolean;
    totalSessions: number;
    onCancel(): void;
    onConfirm(): void;
}

const ModalConfirmDelete: React.StatelessComponent<IPropsModalConfirmDelete> = (props) => {
    const firstText = props.totalSessions === 1 ? 'la sesión' : 'las sesiones';
    const secondText = props.totalSessions === 1 ? 'esta sesión' : 'estas sesiones';
    return (
        <ConsoleModalConfirm
            show={props.show}
            title={`¿Estás seguro que deseas eliminar\n${firstText}?`} >
            <div className={'ModalConfirmDelete'}>
                <Title2 style={{color: "#ee5c7d"}}>
                    <span style={{position: 'relative'}}>
                        <Icon name={'exclamation'} style={
                            {fill: "#ee5c7d", height: 24, width: 24, position: 'absolute', top: 4 }}/>
                    </span>&nbsp;
                    <span style={{paddingLeft: 25}}>Si eliminas {secondText} no podrás revertir esta acción más adelante.</span>
                </Title2>
            </div>
            <ConfirmButtons
                styles={{width: '74%', margin: '60px auto 20px auto'}}
                disabled={props.disabled}
                loading={props.loading}
                cancelText={'No, volver'}
                confirmText={'Si, eliminar'}
                onCancel={props.onCancel}
                onConfirm={props.onConfirm} />
        </ConsoleModalConfirm>
    );
};

export default ModalConfirmDelete;