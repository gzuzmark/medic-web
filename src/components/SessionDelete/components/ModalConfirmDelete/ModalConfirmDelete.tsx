import * as React from 'react';
import ConfirmButtons from "../../../../common/ConfirmButtons/ConfirmButtons";
import ConsoleModalConfirm from "../../../../common/ConsoleModal/ConsoleModalConfirm";
import {Title2} from "../../../../common/ConsoleText";
import './ModalConfirmDelete.scss';


interface IPropsModalConfirmDelete {
    show: boolean;
    styles?: React.CSSProperties;
    disabled: boolean;
    loading: boolean;
    onCancel(): void;
    onConfirm(): void;
}

const ModalConfirmDelete: React.StatelessComponent<IPropsModalConfirmDelete> = (props) => {
    return (
        <ConsoleModalConfirm
            show={props.show}
            title={'¿Está seguro que desea eliminar estos talleres?'} >
            <div className={'ModalConfirmDelete'}>
                <Title2 color="textBold">Si cancelas, se perderá la información actual.</Title2>
                <Title2 color="textBold" style={{fontWeight: 'normal'}}>¿Estás seguro?</Title2>
            </div>
            <ConfirmButtons
                styles={{width: '74%', margin: '60px auto 20px auto'}}
                disabled={props.disabled}
                loading={props.loading}
                cancelText={'No, volver'}
                confirmText={'Si, eliminar'}
                onCancel={props.onCancel}
                onConfirm={props.onConfirm} />
        </ConsoleModalConfirm>
    );
};

export default ModalConfirmDelete;