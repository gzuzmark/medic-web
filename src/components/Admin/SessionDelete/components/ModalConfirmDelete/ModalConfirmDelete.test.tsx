import { shallow } from 'enzyme';
import 'jest-localstorage-mock';
import * as React from 'react';
import ModalConfirmDelete from './ModalConfirmDelete';


describe('ModalConfirmDelete Test',() => {
    let props: any;
    let mountedModalConfirmDelete: any;
    const getComponent = () => {
        if (!mountedModalConfirmDelete) {
            mountedModalConfirmDelete = shallow(
                <ModalConfirmDelete {...props}>Contenido de la Modal</ModalConfirmDelete>
            );
        }
        return mountedModalConfirmDelete;
    };

    beforeEach(() => {
        props = {
            disabled: false,
            loading: false,
            onCancel: () => {
                //
            },
            onConfirm: () => {
                //
            },
            show: true
        };
        mountedModalConfirmDelete = undefined;
    });

    it("render: render ModalConfirmDelete", () => {
        const component = getComponent();
        expect(component).toMatchSnapshot();
    });
});