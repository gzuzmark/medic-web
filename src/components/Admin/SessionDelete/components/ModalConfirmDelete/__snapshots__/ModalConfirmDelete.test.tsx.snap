// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalConfirmDelete Test render: render ModalConfirmDelete 1`] = `
<MentorModalBase
  show={true}
>
  <div
    className="ModalConfirmDelete"
  >
    <div
      className="ModalConfirmDelete_texts"
    >
      <styled.h3
        style={
          Object {
            "margin": "7px 0",
          }
        }
      >
        ¿Estás seguro que deseas eliminar 
        las undefined sesiones
        ?
      </styled.h3>
      <styled.h4
        style={
          Object {
            "margin": "7px 0",
          }
        }
        weight="300"
      >
        Si eliminas 
        estas sesiones
         no podrás revertir esta acción más adelante.
      </styled.h4>
    </div>
    <div
      className="ModalConfirmDelete_buttons"
    >
      <Styled(Button)
        attrs={
          Object {
            "onClick": [Function],
          }
        }
        text="Cancelar"
        type="secondary"
      />
      <Styled(Button)
        attrs={
          Object {
            "onClick": [Function],
          }
        }
        text="Aceptar"
      />
    </div>
  </div>
</MentorModalBase>
`;
